hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -1060,6 +1074,44 @@ public void updateMembers(Collection<MemberInfo> members) {
         }
     }
 
+    private boolean shouldProcessMemberUpdate(Map<Address, MemberImpl> currentMembers,
+                                              Collection<MemberInfo> newMemberInfos) {
+
+        int currentMembersSize = currentMembers.size();
+        int newMembersSize = newMemberInfos.size();
+
+        if (currentMembersSize > newMembersSize) {
```

Comment:
Image cluster size is increasing consequently, which is most likely when the cluster is just forming, in a 10 node cluster if we just receive 9 node member-list we can assume that is old. Even though the cluster size decreased this is not the place we update the member-list for dead member since it just updates the member-list. We have to wait for event to properly declare the node as dead.